aptos-move/aptos-vm-types/src/resource_group_adapter.rs

```diff
@@ -25,20 +26,33 @@ pub enum GroupSizeKind {
 }
 
 impl GroupSizeKind {
-    pub fn from_gas_feature_version(gas_feature_version: u64) -> Self {
-        if gas_feature_version >= 9 {
-            if gas_feature_version >= 12 {
-                GroupSizeKind::AsSum
-            } else {
-                // Keep old caching behavior for replay.
-                GroupSizeKind::AsBlob
-            }
+    pub fn from_gas_feature_version(
+        gas_feature_version: u64,
+        resource_group_charge_as_size_sum_enabled: bool,
```

Comment:
I think here it's mainly to decouple the feature from other gas feature changes and allow easily enabling it on/off (while it can have its own gas feature version, it'd have to be the last one and is trickier). Also, I think it's easier to say & the feature version for delayed fields with this feature version. @igor-aptos knows best.